R/zoop_downloader.R

```diff
@@ -183,10 +195,135 @@ Zoopdownloader <- function(
+      Tryer(n=3, fun=utils::download.file, url=DOP_Meso_URL,
+            destfile=file.path(Data_folder, DOP_Meso_file), mode="wb", method=Download_method)
+    }
+    if (!file.exists(file.path(Data_folder, DOP_trawls_file)) | Redownload_data) {
+      Tryer(n=3, fun=utils::download.file, url=DOP_trawls_URL,
+            destfile=file.path(Data_folder, DOP_trawls_file), mode="wb", method=Download_method)
+    }
+
+
+    # Import the DOP data
+
+    zoo_DOP_Meso<-readr::read_csv(file.path(Data_folder, DOP_Meso_file))
+    zoo_DOP_trawls<-readr::read_csv(file.path(Data_folder, DOP_trawls_file))
+
+    # Tranform from "wide" to "long" format, add some variables,
+    # alter data to match other datasets
+
+    data.list[["DOP_Meso"]] <- zoo_DOP_Meso %>%
+      tidyr::pivot_longer(cols = !ICF_ID, names_to = "DOP_Meso", values_to = "CPUE") %>%
+      dplyr::left_join(zoo_DOP_trawls) %>%
+      dplyr::mutate(Datetime =  lubridate::ymd(as.character(Date), tz = "America/Los_Angeles") + lubridate::hms(as.character(Start_Time)),
+              Source = "DOP", #add variable for data source
+              SizeClass = "Meso") %>%
+      dplyr::filter(!is.na(.data$Mesozooplankton_Volume)) %>% #get rid of environmental variables with no data
+
+      #Select variables we are interested in. I need to check on the latitude/longitude issue with Sam.
+      dplyr::select(.data$Source, .data$Date, .data$Datetime,
+                    Station = .data$Station_Code, Chl = .data$Chl_a, CondSurf = .data$Conductivity, .data$Secchi, .data$SizeClass,
+                    .data$Temperature, Volume = .data$Mesozooplankton_Volume, BottomDepth = .data$Start_Depth,
+                    .data$DOP_Meso, .data$CPUE, .data$Latitude, .data$Longitude, .data$ICF_ID) %>%
+      dplyr::left_join(Crosswalk %>% #Add in Taxnames, Lifestage, and taxonomic info
+                       dplyr::select(.data$DOP_Meso, .data$Lifestage, .data$Taxname, .data$Phylum,
+                                     .data$Class, .data$Order, .data$Family, .data$Genus, .data$Species)%>% #only retain dop codes
+                         dplyr::filter(!is.na(.data$DOP_Meso))%>% #Only retain Taxnames corresponding to EMP codes
+                         dplyr::distinct(),
+                       by="DOP_Meso")%>%
+      dplyr::filter(!is.na(.data$Taxname),#I might not need this step for this dataset, but just in case
+                    !is.na(CPUE)) %>%  #get rid of the lines with "NA" because the critter wasn't counted in this sample.
+      dplyr::mutate(Taxlifestage=paste(.data$Taxname, .data$Lifestage), #create variable for combo taxonomy x life stage
+                    SampleID=paste(.data$Source, .data$Station, .data$Date, .data$ICF_ID), #Create identifier for each sample
+                    BottomDepth=.data$BottomDepth*0.3048,# Convert feet to meters
+                    CondBott = NA, Tide = factor(NA))%>% #needed to prevent error when combining data
```

Comment:
They don't have a tide variable? That's a bummer. I'm going to try some edits to make these changes unneeded.

Reply:
Thanks! Yeah, it is a bummer.